workers/unity/Assets/Playground/Scripts/EntityInitialization/GameObjectInitializationSystem.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections.Generic;
+using Generated.Improbable.Transform;
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Playground
+{
+    /// <summary>
+    ///     Creates a companion gameobject for newly spawned entities according to a prefab definition.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.EntityInitialisationGroup))]
+    internal class GameObjectInitializationSystem : ComponentSystem
+    {
+        public struct AddedEntitiesData
+        {
+            public int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentArray<SpatialOSPrefab> PrefabNames;
+            [ReadOnly] public ComponentDataArray<SpatialOSTransform> Transforms;
+            [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
+            [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> NewlyCreatedEntities;
+        }
+
+        public struct RemovedEntitiesData
+        {
+            public int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
+            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+        }
+
+        [Inject] private AddedEntitiesData addedEntitiesData;
+        [Inject] private RemovedEntitiesData removedEntitiesData;
+
+        private MutableView view;
+        private WorkerBase worker;
+        private Vector3 origin;
+        private readonly ViewCommandBuffer viewCommandBuffer = new ViewCommandBuffer();
+        private EntityGameObjectCreator entityGameObjectCreator;
+        private uint currentHandle;
+        private readonly Dictionary<uint, GameObject> entityGameObjectCache = new Dictionary<uint, GameObject>();
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            worker = WorkerRegistry.GetWorkerForWorld(World);
+            view = worker.View;
+            origin = worker.Origin;
+            entityGameObjectCreator = new EntityGameObjectCreator(World, new Dictionary<string, GameObject>());
+        }
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < addedEntitiesData.Length; i++)
+            {
+                var prefabMapping = PrefabConfig.PrefabMappings[addedEntitiesData.PrefabNames[i].Prefab];
+                var transform = addedEntitiesData.Transforms[i];
+                var entity = addedEntitiesData.Entities[i];
+                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i].EntityId;
+
+                if (!(worker is UnityClient) && !(worker is UnityGameLogic))
+                {
+                    Debug.LogErrorFormat(Errors.UnknownWorkerType, World.Name);
```

Comment:
Should `continue` or `return` or something here.

Reply:
Nice catch! Fixed. :)

Reply:
Please use the log dispatcher here.

Reply:
Done.